plugins/drill/index.tsx

```diff
@@ -0,0 +1,159 @@
+import { Coord, VoxelCoord } from "@latticexyz/utils";
+import { render } from "preact";
+import { useState, useEffect } from "preact/hooks";
+import { Subscription } from "rxjs";
+
+const ID = "drill-plugin-root";
+const DIAMOND_ID = '0x07ada0086470563550001d9d522de93345af1f24d3f49468f1474ef163c5f959';
+
+const {
+  network: {
+    api: { mine, getBlockAtPosition }
+  },
+  noa: {
+    noa: { inputs, container },
+    streams: { playerChunk$ },
+  },
+} = window.layers;
+
+// Create React UI
+const Container = () => {
+  const [disabled, setDisabled] = useState(true);
+  const [visible, setVisible] = useState(false);
+  const [subscription, setSubscription] = useState<Subscription>();
+  const [playerChunk, setPlayerChunk] = useState<Coord>();
+  const [localDiamonds, setLocalDiamonds] = useState<VoxelCoord[]>([]);
+  const [hover, setHover] = useState(false);
+
+  function toggleUI() {
+    setVisible((v) => {
+      if (!container.hasPointerLock && !v) return v;
+      container.setPointerLock(v);
+      return !v;
+    });
+  }
+
+  useEffect(() => {
+    // subscribe to chunk updates
+    if(subscription) subscription.unsubscribe();
+    
+    setSubscription(playerChunk$.subscribe((value) => {
+      if (!playerChunk || value.x != playerChunk.x || value.y != playerChunk.y) {
+        setPlayerChunk(value);
+      }
+    }));
+    
+  }, [playerChunk])
+
+  useEffect(() => {
+    if (!playerChunk) return;
+
+    // when position changes find diamonds on their chunk
+    let diamondCoords = [];
+    for (let y = -20; y <= 50; y++) {
```

Comment:
i don't think diamonds will be in this y range unless you are in a mountain biome

Reply:
That's right, it definitely does a lot of checking "air" blocks in this case when someone is not in a mountain biome. But there is already a case where you stand on top of a mountain, place a diamond at y = 51, and the diamond counter does not pick it up.

Reply:
For this reason, I actually considered _raising_ this to include almost every y value. I just pushed a change where this had circular dependencies and the `useEffect` was firing all the time. Current state is that it will only run once when the player moves from one chunk to another.